src/main/java/it/gov/pagopa/xpay/controller/XPayController.java

```diff
@@ -0,0 +1,72 @@
+package it.gov.pagopa.xpay.controller;
+
+import com.github.javafaker.Bool;
+import it.gov.pagopa.db.repository.TableConfigRepository;
+import it.gov.pagopa.xpay.dto.EsitoXpay;
+import it.gov.pagopa.xpay.dto.XPayAuthRequest;
+import it.gov.pagopa.xpay.dto.XPayAuthResponse;
+import it.gov.pagopa.xpay.dto.XpayError;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.math.BigInteger;
+
+@Validated
+@RestController
+@RequestMapping("/xpay")
+@Log4j2
+public class XPayController {
+
+    @Autowired
+    private TableConfigRepository configRepository;
+
+    private String htmlToReturn = "<html>\n" +
+            "   <head>\n" +
+            "      <title>\\nGestione Pagamento - Autenticazione</title>\n" +
+            "      <script type=\\\"text/javascript\\\" language=\\\"javascript\\\">\\nfunction moveWindow() {\\n    document.tdsFraudForm.submit();\\n}\\n</script>\\n\n" +
+            "   </head>\n" +
+            "   <body>\n" +
+            "      <form name=\\\"tdsFraudForm\\\" action=\\\"mock-psp-xpay-view\\\" method=\\\"POST\\\">\\n<input type=\\\"hidden\\\" name=\\\"action\\\"     value=\\\"fraud\\\">\\n<input type=\\\"hidden\\\" name=\\\"merchantId\\\" value=\\\"31320986\\\">\\n<input type=\\\"hidden\\\" name=\\\"description\\\" value=\\\"7090132540_1663077273191\\\">\\n<input type=\\\"hidden\\\" name=\\\"gdiUrl\\\"      value=\\\"\\\">\\n<input type=\\\"hidden\\\" name=\\\"gdiNotify\\\"   value=\\\"\\\">\\n</form>\n" +
+            "      <script type=\\\"text/javascript\\\">\\n  moveWindow();\\n</script>\\n\n" +
+            "   </body>\n" +
+            "</html>";
+
+    @PostMapping("/ecomm/api/paga/autenticazione3DS")
+    public ResponseEntity<Object> paymentAuthorization(@RequestBody XPayAuthRequest request) {
+
+        log.info("Create AuthResponse for transactioId: " + request.getCodiceTransazione());
+
+        Boolean isMacEqual = checkMac(request.getMac(), request.getCodiceTransazione(), request.getDivisa(), request.getImporto(), request.getTimeStamp());
+
+        XPayAuthResponse xPayAuthResponse = new XPayAuthResponse();
```

Comment:
@DavideAmatoNexi following Nexi documentation https://ecommerce.nexi.it/specifiche-tecniche/servertoserver/pagamento3dsecure/autenticazione.html the response object should contain more parameters, depending on the authorization outcome (OK/KO)